src/Mod/PartDesign/InvoluteGearFeature.py

```diff
@@ -76,28 +75,53 @@ class _InvoluteGear:
-        obj.HighPrecision = True
-        obj.ExternalGear = True
-
+        self._ensure_properties(obj, is_restore=False)
         obj.Proxy = self
 
+    def onDocumentRestored(self, obj):
+        """hook used to migrate older versions of this object"""
+        self._ensure_properties(obj, is_restore=True)
+
+    def _ensure_properties(self, obj, is_restore):
+        def ensure_property(type_, name, doc, default):
+            if not hasattr(obj, name):
+                obj.addProperty(type_, name, "Gear", doc)
+                if callable(default):
+                    setattr(obj, name, default())
+                else:
+                    setattr(obj, name, default)
+
+        ensure_property("App::PropertyInteger", "NumberOfTeeth",
+            doc="Number of gear teeth",
+            default=26)
+        ensure_property("App::PropertyLength", "Modules",
+            doc="Modules of the gear",
+            default="2.5 mm")
+        ensure_property("App::PropertyAngle", "PressureAngle",
+            doc="Pressure angle of gear teeth",
+            default="20 deg")
+        ensure_property("App::PropertyBool", "HighPrecision",
+            doc="True=2 curves with each 3 control points False=1 curve with 4 control points.",
+            default=True)
+        ensure_property("App::PropertyBool", "ExternalGear",
+            doc="True=external Gear False=internal Gear",
+            default=True)
+        ensure_property("App::PropertyFloat", "AddendumCoefficient",
+            doc="The height of the tooth from the pitch circle up to its tip, normalized by the module.",
+            default=lambda: 1.0 if obj.ExternalGear else 0.6)
+        ensure_property("App::PropertyFloat","DedendumCoefficient",
+            doc="The height of the tooth from the pitch circle down to its root, normalized by the module.",
+            default=1.25)
+        ensure_property("App::PropertyFloat","RootFilletCoefficient",
+            doc="The radius of the fillet at the root of the tooth, normalized by the module.",
+            default=lambda: 0.375 if is_restore else 0.38)
 
     def execute(self,obj):
-        #print "_InvoluteGear.execute()"
         w = fcgear.FCWireBuilder()
-        if obj.ExternalGear:
-            involute.CreateExternalGear(w, obj.Modules.Value,obj.NumberOfTeeth, obj.PressureAngle.Value, obj.HighPrecision)
-        else:
-            involute.CreateInternalGear(w, obj.Modules.Value,obj.NumberOfTeeth, obj.PressureAngle.Value, obj.HighPrecision)
+        generator_func = involute.CreateExternalGear if obj.ExternalGear else involute.CreateInternalGear
+        generator_func(w, obj.Modules.Value, obj.NumberOfTeeth, obj.PressureAngle.Value,
+            split=obj.HighPrecision, addCoeff=obj.AddendumCoefficient, dedCoeff=obj.DedendumCoefficient,
+            filletCoeff=obj.RootFilletCoefficient)
         gearw = Part.Wire([o.toShape() for o in w.wire])
         obj.Shape = gearw
         obj.positionBySupport();
```

Comment:
This is not related to your PR but do you mind adding a separate commit to remove this unnecessary semicolon?